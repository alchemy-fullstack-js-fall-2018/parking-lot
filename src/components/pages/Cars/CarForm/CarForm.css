.registerForm {
  display: grid;
  grid-template-columns: 300px 300px;
  margin: 0px 15%;
}
.registerForm label {
  justify-self: end;
  margin: 5px;
}

.registerForm input {
  justify-self: start;
  margin: 5px;
}
.registerForm input[type=text] {
  width: 100%;
}

.registerForm select {
  justify-self: start;
  margin: 5px;
}
