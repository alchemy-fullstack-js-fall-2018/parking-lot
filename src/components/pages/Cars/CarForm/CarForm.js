import React from 'react';
import PropTypes from 'prop-types';

const CarForm = () => {
  return <div>CarForm</div>;
};

CarForm.propTypes = {};

export default CarForm;
