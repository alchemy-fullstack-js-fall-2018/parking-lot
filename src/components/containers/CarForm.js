import { connect } from 'react-redux';
import CarForm from '../pages/cars/CarForm.jsx';
import { registerCar } from '../../state/actions/cars';

const mapStateToProps = () => {
  return {};
};

const mapDispatchToProps = dispatch => ({
  registerCar: (make, model, plate) => dispatch(registerCar(make, model, plate))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CarForm);
