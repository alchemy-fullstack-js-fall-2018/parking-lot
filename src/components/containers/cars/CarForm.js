import { connect } from 'react-redux';
import CarForm from '../../pages/cars/CarForm.jsx';
import { registerCar } from '../../../data/state/actions/cars';

const mapStateToProps = () => {
  return {};
};

const mapDispatchToProps = dispatch => ({
  registerCar: (car) => dispatch(registerCar(car))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CarForm);
