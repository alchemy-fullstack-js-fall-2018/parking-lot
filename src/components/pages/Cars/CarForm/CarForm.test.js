import React from 'react';
import { shallow } from 'enzyme';
import CarForm from './CarForm';

describe('<CarForm />', () => {
  test('renders', () => {
    const wrapper = shallow(<CarForm />);
    expect(wrapper).toMatchSnapshot();
  });
});
