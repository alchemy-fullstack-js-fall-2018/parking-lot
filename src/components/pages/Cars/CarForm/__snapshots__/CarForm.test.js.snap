// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarForm /> renders 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="registerForm"
  >
    <label
      htmlFor="plate"
    >
      Plate
    </label>
    <input
      name="plate"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="make"
    >
      Make
    </label>
    <input
      name="make"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="model"
    >
      Model
    </label>
    <input
      name="model"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <button
    type="submit"
  >
    Register
  </button>
</form>
`;
